[FEATURE] Новая команда - конвертирование изобравжения в ASCII.

idmp152 opened the pull request.

idmp152 committed:
[FEATURE] Новая команда - конвертирование изображения в ASCII.

idmp152 committed:
[FIX] Забыл импортнуть модуль.

IteratorW commented on mrvn_modules/fun_stuff.py:
```diff
@@ -252,6 +254,26 @@ async def execute(self, ctx: CommandContext) -> CommandResult:
+        class ITACommand(Command):
+            async def execute(self, ctx: CommandContext) -> CommandResult:
+                if len(ctx.message.attachments) != 0:
+                    req = requests.get(ctx.message.attachments[0], allow_redirects=True)
```
Стоило бы добавить эксепт на ошибку запроса

IteratorW commented on mrvn_modules/fun_stuff.py:
```diff
@@ -252,6 +254,26 @@ async def execute(self, ctx: CommandContext) -> CommandResult:
+            async def execute(self, ctx: CommandContext) -> CommandResult:
+                if len(ctx.message.attachments) != 0:
+                    req = requests.get(ctx.message.attachments[0], allow_redirects=True)
+                    open('src_image.png', 'wb').write(req.content)
```
Здесь лучше использовать with

IteratorW replied:
Бот асинхронный, могут работать сразу две команды, соответственно этот файл уже может быть. Надо добавить уникальный ID к имени (например, ID сообщения)

IteratorW commented on mrvn_modules/fun_stuff.py:
```diff
@@ -252,6 +254,26 @@ async def execute(self, ctx: CommandContext) -> CommandResult:
+                if len(ctx.message.attachments) != 0:
+                    req = requests.get(ctx.message.attachments[0], allow_redirects=True)
+                    open('src_image.png', 'wb').write(req.content)
+                    img = Image.open("test1.png")
```
Что оно открывает?

IteratorW replied:
А если это вообще не изображение? Нет проверки на то, что юзверь отправляет команде.

idmp152 committed:
[FIX] Выполнены просьбы из ревью.

https://github.com/alfoworks/M.R.V.N./pull/5/files/33e2fb39fca68a0c12bae15560509f0f1bc759c2

IteratorW commented on mrvn_modules/fun_stuff.py:
```diff
@@ -252,6 +254,33 @@ async def execute(self, ctx: CommandContext) -> CommandResult:
+                            pixel = img.getpixel((j, i))
+                            res = res + '**' + symbols[int((pixel*9)/255)] + '**'
+                        res = res + '\n'
+                    os.remove('src_image.png'+ctx.message.id)
```
Надо бы вставить ID не в расширение файла)

IteratorW commented on mrvn_modules/fun_stuff.py:
```diff
@@ -252,6 +254,33 @@ async def execute(self, ctx: CommandContext) -> CommandResult:
+                        req = requests.get(ctx.message.attachments[0], allow_redirects=True)
+                    except RequestException:
+                        return CommandResult.error("Ошибка запроса!")
+                    with open('src_image.png_'+ctx.message.id, 'wb') as f:
```
Надо бы вставить ID не в расширение файла)

idmp152 committed:
[FIX] Пофикшено добавление ID

IteratorW committed:
Merge branch 'master' into master
